app/src/main/java/com/android/streetworkapp/ui/train/TrainHubScreen.kt

```diff
@@ -0,0 +1,214 @@
+package com.android.streetworkapp.ui.train
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.lazy.grid.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.sample.R
+import com.android.streetworkapp.model.user.UserViewModel
+import com.android.streetworkapp.model.workout.WorkoutViewModel
+import com.android.streetworkapp.ui.navigation.NavigationActions
+import com.android.streetworkapp.ui.theme.ColorPalette.BORDER_COLOR
+import com.android.streetworkapp.ui.theme.ColorPalette.INTERACTION_COLOR_DARK
+
+@Composable
+fun TrainHubScreen(
+    navigationActions: NavigationActions,
+    workoutViewModel: WorkoutViewModel,
+    userViewModel: UserViewModel,
+    paddingValues: PaddingValues = PaddingValues(0.dp)
+) {
+  var selectedType by remember { mutableStateOf<String?>(null) }
+  var selectedActivity by remember { mutableStateOf<Pair<String, Boolean>?>(null) }
+
+  val currentUser by userViewModel.currentUser.collectAsState()
+  currentUser?.uid?.let { workoutViewModel.getOrAddWorkoutData(it) }
+
+  BoxWithConstraints {
+    val screenWidth = maxWidth
+    val screenHeight = maxHeight
+
+    val buttonSize =
+        ButtonSize(
+            width = if (screenWidth < 360.dp) 80.dp else 100.dp,
+            height = if (screenHeight < 600.dp) 70.dp else 100.dp,
+            spacing = if (screenWidth < 360.dp) 4.dp else 8.dp)
+
```

Comment:
Is it wanted for the height and width to be different ? I.e 80x70 or 100x100 (Should it be 80x80 ? Or 70x70 ?)

Reply:
To my knowledge, there is no android cell phone with a square screen, with such a small resolution, and even if there were, the icons would be centered with a small spacing.

Reply:
My question was simply to make sure that you wanted the buttonSize to be 80x70 and 100x100. I have no issue with the spacing. But why would you handle this edge case if to your knowlege such screen does not exist ?

Reply:
I just adapt my button size depending the screen size but the 80x70 and 100x100 is not explicitly handle.

Reply:
I was simply bugged out by your choice of button size being set to 80x70 while having a square button for other resolution.